test/lib/convert-forms/index.spec.js

```diff
@@ -0,0 +1,158 @@
+const { expect } = require('chai');
+const sinon = require('sinon');
+const rewire = require('rewire');
+
+const fs = require('../../../src/lib/sync-fs');
+const path = require('path');
+const log = require('../../../src/lib/log');
+const { LEVEL_NONE } = log;
+
+const XLS2XFORM = path.join(__dirname, '..', '..', '..', 'bin', 'xls2xform-medic');
+
+describe('convert-forms', () => {
+  let convertForms;
+  let mockExec;
+  let withMocks;
+  beforeEach(() => {
+    mockExec = sinon.stub();
+    sinon.stub(log, 'warn');
+    convertForms = rewire('./../../../src/lib/convert-forms');
+    withMocks = cb => convertForms.__with__({
+      exec: mockExec,
+      fixXml: sinon.stub(),
+      getHiddenFields: sinon.stub(),
+    })(cb);
+
+    sinon.stub(fs, 'readdir').returns(['a.xml', 'b.xlsx', 'c.xlsx']);
+    sinon.stub(fs, 'exists').returns(true);
+    sinon.stub(fs, 'readJson').returns({});
+  });
+  afterEach(sinon.restore);
+
+  describe('pyxform execution fails', () => {
+    const message = 'Python is not installed.';
+    [
+      new Error(message),
+      { message },
+      message
+    ].forEach(error => {
+      it('throws error with error message', () => withMocks(async () => {
+        mockExec.returns(Promise.reject(error));
+
+        await expect(convertForms.execute('./path', 'app')).to.be.rejectedWith(
+          `There was a problem executing xls2xform. Make sure you have Python 3.10+ installed.\n${message}`
```

Comment:
Is there a way that we can avoid this callback test model? I believe that this fails miserably with a timeout if assertions fail? 

An option is to use `__set__` for each replaced property and just call `convertForms` as normal. 

```
convertForms = rewire('./../../../src/lib/convert-forms');
convertForms.__set__('exec', mockExec);
convertForms.__set__('fixXml', );
convertForms.__set__('execgetHiddenFields, sinon.stub());
....
await convertForms(...)
```

Reply:
Ahh, got it. Yeah, I can def get rid of that...